lib/pbench/test/unit/server/query_apis/test_datasets_detail.py

```diff
@@ -265,43 +259,38 @@ def test_nonunique_query(
 
         index = self.build_index(server_config, ("2020-08", "2020-09", "2020-10"))
         response = query_api(
-            json, index, HTTPStatus.BAD_REQUEST, server_config, json=response_payload
+            "/datasets/detail",
+            "/_search?ignore_unavailable=true",
+            json,
+            index,
+            HTTPStatus.BAD_REQUEST,
+            json=response_payload,
         )
         assert response.json["message"].find("Too many hits for a unique query") != -1
 
     @pytest.mark.parametrize(
         "exceptions",
         (
             {
-                "exception": make_http_exception(HTTPStatus.BAD_REQUEST),
+                "exception": requests.exceptions.ConnectionError(),
                 "status": HTTPStatus.BAD_GATEWAY,
             },
             {
-                "exception": requests.exceptions.ConnectionError,
-                "status": HTTPStatus.BAD_GATEWAY,
-            },
-            {
-                "exception": requests.exceptions.Timeout,
+                "exception": requests.exceptions.Timeout(),
                 "status": HTTPStatus.GATEWAY_TIMEOUT,
             },
             {
-                "exception": requests.exceptions.InvalidURL,
+                "exception": requests.exceptions.InvalidURL(),
                 "status": HTTPStatus.INTERNAL_SERVER_ERROR,
             },
-            {"exception": Exception, "status": HTTPStatus.INTERNAL_SERVER_ERROR},
+            {"exception": Exception(), "status": HTTPStatus.INTERNAL_SERVER_ERROR},
```

Comment:
Wouldn't it make for a better test if you picked a concrete exception or a subclass of `Exception` which would then be caught by a match to `Exception`?

Reply:
Again, the use of `Exception` rather than an arbitrary subclass predates this PR so I'm not too inclined to over-analyze. However we _do_ want to be sure that we test any `isinstance(e, Exception)` so I'm not sure any other test is truly appropriate short of iterating through all/many subclasses.

Reply:
Let me count the number of instances, "one...two...many".  OK, so we need to test three.  ;-)

(Actually, two is probably enough.  Really, I think one will do, so long as it's not `Exception` itself.  :-D )